src/main/scala/com.snowplowanalytics.snowplow.storage.kinesis/s3/serializers/ISerializer.scala

```diff
@@ -26,5 +35,21 @@ case class SerializationResult(namedStreams: List[NamedStream], results: List[Em
  * Shared interface for all serializers
  */
 trait ISerializer {
-  def serialize(records: List[ EmitterInput ], baseFilename: String): SerializationResult
+  def serialize(records: List[EmitterInput], baseFilename: String): SerializationResult
+
+  def serializeRecord[T](
+    record: RawRecord,
+    serializer: T,
+    serialize: T => Unit
+  ): ValidatedRecord =
+    try {
+      serialize(serializer)
+      record.success
+    } catch {
+      case e: IOException =>
+        val base64Record = new String(Base64.encodeBase64(record), "UTF-8")
+        FailedRecord(List(s"Error writing raw event to output stream: [$e]"), base64Record).failure
+      case NonFatal(e) =>
+        FailedRecord(List(s"Error writing raw event to output stream: [$e]"), "").failure
```

Comment:
Is there a reason for the removal of:

```scala
// Need to log OutOfMemoryErrors
case t: Throwable => {
  log.error("Error writing raw stream to output stream", t)
  throw t
}
```

According to [this](http://www.scala-lang.org/api/2.9.3/scala/util/control/NonFatal$.html) it will not match on an OOM error.